content/blog/reflections-of-a-pulumi-intern/index.md

```diff
@@ -0,0 +1,66 @@
+
+## The First Week
+
+I had very little idea what the internship would look like. In addition to it being virtual, it was my first software engineering gig, so I had no prior experience to base it off. I wasn't sure what my role would be—would I just be assigned a single project and be told to work on it? Would I have a mentor? How does the process of submitting code look like? Fortunately, these questions were answered fairly quickly.
+
+For the first week, the other interns and I converted Pulumi [examples](https://github.com/pulumi/examples/) from one language to another. TypeScript seemed to be the most common language, so I converted a few from that language to some of the relatively less popular languages, such as Python and Go. I thought that this was an excellent way to help us interns ramp up, since it allowed us to get some baseline experience with Pulumi and the various cloud providers and programming languages it supports. By Friday, I was pretty familiar with how to navigate the various Pulumi repositories and documentation, which was a huge benefit as I launched into my project.
+
+## My Project: `crd2pulumi`
+
+On the following Monday, I joined a Zoom meeting with [Levi](https://www.pulumi.com/blog/author/levi-blackstone/) and my mentor [Justin](https://www.pulumi.com/blog/author/justin-vanpatten/). They had written up a preliminary design doc on generating SDK types for Kubernetes CustomResources, which would be my project for the summer. I had little experiences with Kubernetes, so I couldn't fully conceptualize what the end product would look like until a few days later.
+
+Basically, you can define your own fields of a Kubernetes CustomResource using [CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/) (CRDs). Pulumi already allows you to create CustomResources using the [apiextensions](https://www.pulumi.com/docs/reference/pkg/kubernetes/apiextensions/) package. However there's no way to inherently know the exact fields of a CustomResource, since it's set by some CRD. Therefore we have to set the CustomResource fields to some generic type, such as `map[string]interface{}` in Go or `pulumi.Input<any>` in TypeScript. This makes filling out the fields for for complex CRDs such as cert-manager or Istio very annoying, since they contain thousands of nested properties. Being able to leverage IDE type-checking and auto-complete makes the process much quicker and less error-prone.
```

Comment:
Update link to use `relref`

```suggestion
Basically, you can define your own fields of a Kubernetes CustomResource using [CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/) (CRDs). Pulumi already allows you to create CustomResources using the [apiextensions]({{< relref "/docs/reference/pkg/kubernetes/apiextensions" >}}) package. However there's no way to inherently know the exact fields of a CustomResource, since it's set by some CRD. Therefore we have to set the CustomResource fields to some generic type, such as `map[string]interface{}` in Go or `pulumi.Input<any>` in TypeScript. This makes filling out the fields for for complex CRDs such as cert-manager or Istio very annoying, since they contain thousands of nested properties. Being able to leverage IDE type-checking and auto-complete makes the process much quicker and less error-prone.
```